src/main/java/core/basesyntax/factory/Factory.java

```diff
@@ -0,0 +1,35 @@
+package core.basesyntax.factory;
+
+import core.basesyntax.dao.BetDao;
+import core.basesyntax.dao.BetDaoImpl;
+import core.basesyntax.dao.UserDao;
+import core.basesyntax.dao.UserDaoImpl;
+import core.basesyntax.exceptions.NoAnnotationException;
+import core.basesyntax.lib.Dao;
+
+public class Factory {
+    private static BetDao betDao;
+    private static UserDao userDao;
+
+    public static BetDao getBetDao() {
+        if (betDao == null) {
+            if (BetDaoImpl.class.isAnnotationPresent(Dao.class)) {
+                betDao = new BetDaoImpl();
+            } else {
+                throw new NoAnnotationException("Annotation was not found.");
```

Comment:
According to mentors' reviews, I see that it's better to replace this check into Injector class